### PR TITLE
#1222 Remediating logging calls.

### PR DESCRIPTION
**Pull request by kalbfled**
# Description

The most important change in this PR is updating requirements.txt to use the newest commit in the master branch of notification-utils to incorporate the fix that eliminates all of the logged KeyError, AttributeError, etc. exceptions.  The cause of those exceptions was an improper implementation of custom log formatters.  See [notification-utils PR 118](https://github.com/department-of-veterans-affairs/notification-utils/pull/118).

Before I figured out the formatter problem, I proceeded under the assumption that the errors stemmed from incorrect string substitution in the logging message.  These changes primarily contain tweaks to logging statements that I made while auditing the code.  Most of the changes are switching from "format" based string substitution to the C-style %-based substitution, which is appropriate for logging.

#1222

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Before merging notification-utils PR 118, this PR used the commit hash of 118's branch, rather than master.  I deployed this branch to Dev, sent myself an SMS, and Cris run the regression suite against Dev.  We did not observe any errors.  Subsequently, I merged the utils PR and updated these changes to use the utils master branch.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/clients/sms/firetext.py [hunk 1]
```diff
@@ -57,8 +57,7 @@ def get_name(self):
     def record_outcome(self, success, response):
         status_code = response.status_code if response else 503
 
-        log_message = "API {} request {} on {} response status_code {}".format(
-            "POST",
+        log_message = "API POST request {} on {} response status_code {}".format(
```
Comment (cris-oddball) [inline app/clients/sms/firetext.py:60]: we should be able to use an %s with am if/else condition, any reason why we are not?

File: app/clients/sms/firetext.py [hunk 2]
```diff
@@ -103,6 +102,6 @@ def send_sms(self, to, content, reference, sender=None, **kwargs):
             raise FiretextClientResponseException(response=e.response, exception=e)
         finally:
             elapsed_time = monotonic() - start_time
-            self.current_app.logger.info("Firetext request for {} finished in {}".format(reference, elapsed_time))
+            self.current_app.logger.info("Firetext request for %s finished in %f", reference, elapsed_time)
```
Comment (cris-oddball) [inline app/clients/sms/firetext.py:105]: we should be able to use an %s with am if/else condition, any reason why we are not?

**Review by cris-oddball — CHANGES_REQUESTED** [review]

**Review by jakehova — APPROVED** [review]

File: app/clients/sms/firetext.py (hunk 1 above)
Comment (kalbfled) [reply app/clients/sms/firetext.py:60]: %s should only be used inside logging statements.
https://docs.python.org/3.8/howto/logging.html#logging-variable-data

File: app/clients/sms/firetext.py (hunk 2 above)
Comment (kalbfled) [reply app/clients/sms/firetext.py:105]: I don't follow here.  What is the condition?

File: app/clients/sms/firetext.py (hunk 1 above)
Comment (cris-oddball) [reply app/clients/sms/firetext.py:60]: sorry, got thrown off by this being a log_message. Ideally, an f-string should be used for consistency?

**Review by cris-oddball — APPROVED** [review]

**Pull request merged**